python/mxnet/gluon/parameter.py

```diff
@@ -548,6 +588,45 @@ def get(self, name, **kwargs):
                     setattr(param, k, v)
         return param
 
+    def get_constant(self, name, value=None):
```

Comment:
When developing for a language model with weight drop, we took the approach of adding a new type of Parameter. While using the new WeightDropParameter, we found it quite difficult as user to properly change the type of parameter in the ParameterDict. Since this is a common use case too, should we be open to adding a getter function for such parameter as well? Or should we consider a different pattern?

Reply:
I think I've met some similar problems implementing the weight normalization, which also transforms the parameters.

Reply:
So the pattern here is that we need parameter transformation, and it can be done through overriding the `data()` (and maybe `grad()`) method. This currently requires inheriting the Parameter class, and there is no existing method in casting the parameter type. So in this case, what's needed is a way to cast the parameter.

One alternative is to provide a place to plug in transformation functions in `data()` and `grad()`.